proteus/equivalent_polynomials_coefficients_quad.h

```diff
@@ -0,0 +1,247 @@
+#ifndef EQUIVALENT_POLYNOMIALS_COEFFICIENTS_QUAD_H
+#define EQUIVALENT_POLYNOMIALS_COEFFICIENTS_QUAD_H
+
+namespace equivalent_polynomials
+{
+  template<int nP>
+  inline void _calculate_b(double theta01, double theta02, double theta31, double theta32, 
+                           double phi0, double phi1, double phi2, double phi3, 
+                           double* b_H, double* b_ImH, double* b_D);
+
+  template<>
+inline void _calculate_b<1>(double theta01, double theta02, double theta31, double theta32,
+                                double phi0, double phi1, double phi2, double phi3,
+                                double* b_H, double* b_ImH, double* b_D)
+  {
+    b_H[0] = (1.0/6.0)*(-4*std::pow(theta02, 4)*std::pow(theta31, 2) + 8*std::pow(theta02, 4)*theta31 - 4*std::pow(theta02, 4) + 2*std::pow(theta02, 3)*std::pow(theta31, 2) - 4*std::pow(theta02, 3)*theta31 + 2*std::pow(theta02, 3)*std::sqrt((2*std::pow(theta02, 2)*std::pow(theta31, 2) - 4*std::pow(theta02, 2)*theta31 + 2*std::pow(theta02, 2) + std::pow(theta31, 2) - 2*theta31 + 1)/(std::pow(theta31, 2) - 2*theta31 + 1))*std::sqrt(2*std::pow(theta02, 2) + 1)*(std::pow(theta31, 2) - 2*theta31 + 1) + 2*std::pow(theta02, 3) - 4*std::pow(theta02, 2)*std::pow(theta31, 2) + 8*std::pow(theta02, 2)*theta31 - 4*std::pow(theta02, 2) + theta02*std::pow(theta31, 2) - 2*theta02*theta31 + theta02 - std::pow(theta31, 2) + 2*theta31 + (theta02*(theta01*theta31 - theta01 - theta31 + 1) + theta31*(theta02*theta32 - theta02 - theta32 + 1))*(4*std::pow(theta02, 4)*theta31 - 4*std::pow(theta02, 4) + 4*std::pow(theta02, 2)*theta31 - 4*std::pow(theta02, 2) + theta31 - 1) - 1)/(4*std::pow(theta02, 4)*theta31 - 4*std::pow(theta02, 4) + 4*std::pow(theta02, 2)*theta31 - 4*std::pow(theta02, 2) + theta31 - 1);
```

Comment:
where does this monstrous formula come from?

Reply:
It's generated from scripts/gen_eqp_quad.py. These are the integrals the polynomials basis functions time the Heaviside function (or equivalently the integrals of the polynomials on cut cells).